<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2 Omic data analysis: types of implemented analyses | Privacy protected federated omic data analysis in multi-center studies with DataSHIELD</title>
  <meta name="description" content="2 Omic data analysis: types of implemented analyses | Privacy protected federated omic data analysis in multi-center studies with DataSHIELD" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="2 Omic data analysis: types of implemented analyses | Privacy protected federated omic data analysis in multi-center studies with DataSHIELD" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2 Omic data analysis: types of implemented analyses | Privacy protected federated omic data analysis in multi-center studies with DataSHIELD" />
  
  
  



<meta name="date" content="2022-01-27" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="introduction.html"/>
<link rel="next" href="genome-wide-association-study-gwas-analysis.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script>
  function unrolltab(evt, divName, tabName) {
    var i, tabcontent, tablinks;
    tabcontent = document.getElementById(divName).getElementsByClassName("tabcontent");
    for (i = 0; i < tabcontent.length; i++) {
      tabcontent[i].style.display = "none";
    }
    tablinks = document.getElementById(divName).getElementsByClassName("tablinks");
    for (i = 0; i < tablinks.length; i++) {
      tablinks[i].className = tablinks[i].className.replace(" active", "");
    }
    document.getElementById(tabName).style.display = "block";
    evt.currentTarget.className += " active";
  }
</script>

<script>
  document.addEventListener("DOMContentLoaded", function(event) { 
    document.getElementById("gwas_results_1").click();
    document.getElementById("gwas_plot_1").click();
    document.getElementById("manhattan-plots-tab").click();
    document.getElementById("locuszoom-plots-tabs").click();
 });
</script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="" target="blank">OmicSHIELD</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="part"><span><b>I Preamble</b></span></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#materials-to-read-beforehand"><i class="fa fa-check"></i><b>1.1</b> Materials to read beforehand</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#what-are-resources-a-very-simple-explanation-without-any-technicalities"><i class="fa fa-check"></i><b>1.2</b> What are “resources”: A very simple explanation without any technicalities</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#capabilities-of-omicshield"><i class="fa fa-check"></i><b>1.3</b> Capabilities of OmicSHIELD</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="Omic.html"><a href="Omic.html"><i class="fa fa-check"></i><b>2</b> Omic data analysis: types of implemented analyses</a></li>
<li class="part"><span><b>II Workflows</b></span></li>
<li class="chapter" data-level="3" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html"><i class="fa fa-check"></i><b>3</b> Genome wide association study (GWAS) analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#single-cohort"><i class="fa fa-check"></i><b>3.1</b> Single cohort</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#connection-to-the-opal-server"><i class="fa fa-check"></i><b>3.1.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="3.1.2" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#assign-the-vcf-resources"><i class="fa fa-check"></i><b>3.1.2</b> Assign the VCF resources</a></li>
<li class="chapter" data-level="3.1.3" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#assign-the-phenotypes"><i class="fa fa-check"></i><b>3.1.3</b> Assign the phenotypes</a></li>
<li class="chapter" data-level="3.1.4" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#vcf_merge"><i class="fa fa-check"></i><b>3.1.4</b> Merge VCF (genotype) and phenotype information</a></li>
<li class="chapter" data-level="3.1.5" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#performing-the-gwas"><i class="fa fa-check"></i><b>3.1.5</b> Performing the GWAS</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#multi-cohorts"><i class="fa fa-check"></i><b>3.2</b> Multi cohorts</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#connection-to-the-opal-server-1"><i class="fa fa-check"></i><b>3.2.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="3.2.2" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#multi-vcf"><i class="fa fa-check"></i><b>3.2.2</b> Assign the VCF resources</a></li>
<li class="chapter" data-level="3.2.3" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#assign-the-phenotypes-1"><i class="fa fa-check"></i><b>3.2.3</b> Assign the phenotypes</a></li>
<li class="chapter" data-level="3.2.4" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#merge-vcf-genotype-and-phenotype-information"><i class="fa fa-check"></i><b>3.2.4</b> Merge VCF (genotype) and phenotype information</a></li>
<li class="chapter" data-level="3.2.5" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#performing-the-gwas-1"><i class="fa fa-check"></i><b>3.2.5</b> Performing a meta - GWAS</a></li>
<li class="chapter" data-level="3.2.6" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#gwas-plots"><i class="fa fa-check"></i><b>3.2.6</b> GWAS Plots</a></li>
<li class="chapter" data-level="3.2.7" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#meta-level-quality-control"><i class="fa fa-check"></i><b>3.2.7</b> Meta-level quality control</a></li>
<li class="chapter" data-level="3.2.8" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#meta-analysis"><i class="fa fa-check"></i><b>3.2.8</b> Meta-analysis</a></li>
<li class="chapter" data-level="3.2.9" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#ultra-fast-gwas"><i class="fa fa-check"></i><b>3.2.9</b> Ultra fast pooled GWAS analysis</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#post-omics"><i class="fa fa-check"></i><b>3.3</b> Post-omic analysis</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#enrichment-analysis"><i class="fa fa-check"></i><b>3.3.1</b> Enrichment analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html"><i class="fa fa-check"></i><b>4</b> Polygenic risk scores</a>
<ul>
<li class="chapter" data-level="4.1" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#definition"><i class="fa fa-check"></i><b>4.1</b> Definition</a></li>
<li class="chapter" data-level="4.2" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#study-configuration"><i class="fa fa-check"></i><b>4.2</b> Study configuration</a></li>
<li class="chapter" data-level="4.3" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#source-of-polygenic-scores"><i class="fa fa-check"></i><b>4.3</b> Source of polygenic scores</a></li>
<li class="chapter" data-level="4.4" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#connection-to-the-opal-server-2"><i class="fa fa-check"></i><b>4.4</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="4.5" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#assign-the-vcf-resources-1"><i class="fa fa-check"></i><b>4.5</b> Assign the VCF resources</a></li>
<li class="chapter" data-level="4.6" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#calculate-the-prs"><i class="fa fa-check"></i><b>4.6</b> Calculate the PRS</a>
<ul>
<li class="chapter" data-level="4.6.1" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#pgs-catalog-example"><i class="fa fa-check"></i><b>4.6.1</b> PGS Catalog Example</a></li>
<li class="chapter" data-level="4.6.2" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#custom-prs_table-example"><i class="fa fa-check"></i><b>4.6.2</b> Custom <code>prs_table</code> example</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#study-of-the-prs-results"><i class="fa fa-check"></i><b>4.7</b> Study of the PRS results</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html"><i class="fa fa-check"></i><b>5</b> Differential gene expression (DGE) analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#connection-to-the-opal-server-3"><i class="fa fa-check"></i><b>5.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="5.2" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#assign-the-rse-resource"><i class="fa fa-check"></i><b>5.2</b> Assign the RSE resource</a></li>
<li class="chapter" data-level="5.3" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#inspect-the-rse"><i class="fa fa-check"></i><b>5.3</b> Inspect the RSE</a></li>
<li class="chapter" data-level="5.4" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#pre-processing-for-rnaseq-data"><i class="fa fa-check"></i><b>5.4</b> Pre-processing for RNAseq data</a></li>
<li class="chapter" data-level="5.5" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#dge-analysis"><i class="fa fa-check"></i><b>5.5</b> DGE analysis</a></li>
<li class="chapter" data-level="5.6" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#surrogate-variable-analysis"><i class="fa fa-check"></i><b>5.6</b> Surrogate variable analysis</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html"><i class="fa fa-check"></i><b>6</b> Epigenome-wide association analysis (EWAS)</a>
<ul>
<li class="chapter" data-level="6.1" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#initial-steps-for-all-use-cases"><i class="fa fa-check"></i><b>6.1</b> Initial steps for all use cases</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#connection-to-the-opal-server-4"><i class="fa fa-check"></i><b>6.1.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="6.1.2" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#assign-the-expressionset-resource"><i class="fa fa-check"></i><b>6.1.2</b> Assign the <code>ExpressionSet</code> resource</a></li>
<li class="chapter" data-level="6.1.3" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#ewas-inspect"><i class="fa fa-check"></i><b>6.1.3</b> Inspect the <code>ExpressionSet</code></a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#single-cpg-pooled-analysis"><i class="fa fa-check"></i><b>6.2</b> Single CpG pooled analysis</a></li>
<li class="chapter" data-level="6.3" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#multiple-cpgs-pooled-analysis"><i class="fa fa-check"></i><b>6.3</b> Multiple CpGs pooled analysis</a></li>
<li class="chapter" data-level="6.4" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#full-genome-meta-analysis"><i class="fa fa-check"></i><b>6.4</b> Full genome meta-analysis</a></li>
<li class="chapter" data-level="6.5" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#adjusting-for-surrogate-variables"><i class="fa fa-check"></i><b>6.5</b> Adjusting for surrogate variables</a></li>
</ul></li>
<li class="part"><span><b>III Technical details</b></span></li>
<li class="chapter" data-level="7" data-path="gwas.html"><a href="gwas.html"><i class="fa fa-check"></i><b>7</b> GWAS</a></li>
<li class="chapter" data-level="8" data-path="internal-structure-of-the-prs-functionality.html"><a href="internal-structure-of-the-prs-functionality.html"><i class="fa fa-check"></i><b>8</b> Internal structure of the PRS functionality</a>
<ul>
<li class="chapter" data-level="8.1" data-path="internal-structure-of-the-prs-functionality.html"><a href="internal-structure-of-the-prs-functionality.html#client"><i class="fa fa-check"></i><b>8.1</b> Client</a></li>
<li class="chapter" data-level="8.2" data-path="internal-structure-of-the-prs-functionality.html"><a href="internal-structure-of-the-prs-functionality.html#server-resolver"><i class="fa fa-check"></i><b>8.2</b> Server resolver</a></li>
<li class="chapter" data-level="8.3" data-path="internal-structure-of-the-prs-functionality.html"><a href="internal-structure-of-the-prs-functionality.html#server-analysis"><i class="fa fa-check"></i><b>8.3</b> Server analysis</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="compression-of-gds-files-vs.-performance.html"><a href="compression-of-gds-files-vs.-performance.html"><i class="fa fa-check"></i><b>9</b> Compression of GDS files vs. performance</a></li>
<li class="part"><span><b>IV Appendix</b></span></li>
<li class="chapter" data-level="10" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>10</b> Session info</a></li>
<li class="chapter" data-level="11" data-path="contributos.html"><a href="contributos.html"><i class="fa fa-check"></i><b>11</b> Contributos</a></li>
<li class="part"><span><b>V References</b></span></li>
<li class="chapter" data-level="12" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>12</b> References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Privacy protected federated omic data analysis in multi-center studies with DataSHIELD</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="Omic" class="section level1" number="2">
<h1><span class="header-section-number">2</span> Omic data analysis: types of implemented analyses</h1>
<p>The Figure <a href="Omic.html#fig:opalOmic">2.1</a> describes the different types of ’omic association analyses that can be performed using DataSHIELD client functions implemented in the <em><a href="https://github.com/isglobal-brge/dsOmicsClient">dsOmicsClient</a></em> package. Basically, data (’omic and phenotypes/covariates) can be stored in different sites (http, ssh, AWS S3, local, …) and are managed with Opal through the <em><a href="https://github.com/obiba/resourcer">resourcer</a></em> package and their extensions implemented in <em><a href="https://github.com/isglobal-brge/dsOmics">dsOmics</a></em>.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:opalOmic"></span>
<img src="fig/dsOmics_A.jpg" alt="Non-disclosive omic data analysis with DataSHIELD and Bioconductor. The figure illustrates how the `resourcer` package is used to get access to omic data through the Opal servers. Then DataSHIELD is used in the client side to perform non-disclosive data analyses." width="744" />
<p class="caption">
Figure 2.1: Non-disclosive omic data analysis with DataSHIELD and Bioconductor. The figure illustrates how the <code>resourcer</code> package is used to get access to omic data through the Opal servers. Then DataSHIELD is used in the client side to perform non-disclosive data analyses.
</p>
</div>
<p>Then, <code>dsOmicsClient</code> package allows different types of analyses: pooled and meta-analysis. Both methods are based on fitting different Generalized Linear Models (GLMs) for each feature when assesing association between ’omic data and the phenotype/trait/condition of interest. Of course, non-disclosive ’omic data analysis from a single study can also be performed.</p>
<p>The <strong>pooled approach</strong> (Figure <a href="Omic.html#fig:omicAnal1">2.2</a>) is recommended when the user wants to analyze ’omic data from different sources and obtain results as if the data were located in a single computer. It should be noted that this can be very time consuming when analyzing multiple features since it calls a base function in DataSHIELD (<code>ds.glm</code>) repeatedly. It also cannot be recommended when data are not properly harmonized (e.g. gene expression normalized using different methods, GWAS data having different platforms, …). Furthermore when it is necesary to remove unwanted variability (for transcriptomic and epigenomica analysis) or control for population stratification (for GWAS analysis), this approach cannot be used since we need to develop methods to compute surrogate variables (to remove unwanted variability) or PCAs (to to address population stratification) in a non-disclosive way.</p>
<p>The <strong>meta-analysis approach</strong> Figure <a href="Omic.html#fig:omicAnal2">2.3</a> overcomes the limitations raised when performing pooled analyses. First, the computation issue is addressed by using scalable and fast methods to perform data analysis at whole-genome level at each location The transcriptomic and epigenomic data analyses make use of the widely used <em><a href="https://bioconductor.org/packages/3.12/limma">limma</a></em> package that uses <code>ExpressionSet</code> or <code>RangedSummarizedExperiment</code> Bioc infrastructures to deal with ’omic and phenotypic (e.g covariates). The genomic data are analyzed using <em><a href="https://bioconductor.org/packages/3.12/GWASTools">GWASTools</a></em> and <em><a href="https://bioconductor.org/packages/3.12/GENESIS">GENESIS</a></em> that are designed to perform quality control (QC) and GWAS using GDS infrastructure.</p>
<p>Next, we describe how both approaches are implemented:</p>
<ul>
<li><strong>Pooled approach:</strong> Figure <a href="Omic.html#fig:omicAnal1">2.2</a> illustrate how this analysis is performed. This corresponds to generalized linear models (glm) on data from single or multiple sources. It makes use of <code>ds.glm()</code> function which is a DataSHIELD function that uses an approach that is mathematically equivalent to placing all individual-level data froma all sources in one central warehouse and analysing those data using the conventional <code>glm()</code> function in R. The user can select one (or multiple) features (i.e., genes, transcripts, CpGs, SNPs, …)</li>
</ul>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:omicAnal1"></span>
<img src="fig/dsOmics_B.jpg" alt="Non-disclosive omic data analysis with DataSHIELD and Bioconductor. The figure illustrates how to perform single pooled omic data analysis. The analyses are performed by using a generalized linear model (glm) on data from one or multiple sources. It makes use of `ds.glm()`, a DataSHIELD function, that uses an approach that is mathematically equivalent to placing all individual-level data from all sources in one central warehouse and analysing those data using the conventional `glm()` function in R." width="619" />
<p class="caption">
Figure 2.2: Non-disclosive omic data analysis with DataSHIELD and Bioconductor. The figure illustrates how to perform single pooled omic data analysis. The analyses are performed by using a generalized linear model (glm) on data from one or multiple sources. It makes use of <code>ds.glm()</code>, a DataSHIELD function, that uses an approach that is mathematically equivalent to placing all individual-level data from all sources in one central warehouse and analysing those data using the conventional <code>glm()</code> function in R.
</p>
</div>
<ul>
<li><strong>Meta-analysis:</strong> Figure <a href="Omic.html#fig:omicAnal2">2.3</a> illustrate how this analysis is performed. This corresponds to performing a genome-wide analysis at each location using functions that are specifically design for that purpose and that are scalable. Then the results from each location can be meta-analyzed using methods that meta-analyze either effect sizes or p-values.</li>
</ul>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:omicAnal2"></span>
<img src="fig/dsOmics_C.jpg" alt="Non-disclosive omic data analysis with DataSHIELD and Bioconductor. The figure illustrates how to perform anlyses at genome-wide level from one or multiple sources. It runs standard Bioconductor functions at each server independently to speed up the analyses and in the case of having multiple sources, results can be meta-analyzed uning standar R functions." width="526" />
<p class="caption">
Figure 2.3: Non-disclosive omic data analysis with DataSHIELD and Bioconductor. The figure illustrates how to perform anlyses at genome-wide level from one or multiple sources. It runs standard Bioconductor functions at each server independently to speed up the analyses and in the case of having multiple sources, results can be meta-analyzed uning standar R functions.
</p>
</div>
</div>



</div>
            </section>

          </div>
        </div>
      </div>
<a href="introduction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="genome-wide-association-study-gwas-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["OmicSHIELD_bookdown.pdf", "OmicSHIELD_bookdown.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
