<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4 Polygenic risk scores | Privacy protected federated omic data analysis in multi-center studies with DataSHIELD</title>
  <meta name="description" content="4 Polygenic risk scores | Privacy protected federated omic data analysis in multi-center studies with DataSHIELD" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="4 Polygenic risk scores | Privacy protected federated omic data analysis in multi-center studies with DataSHIELD" />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4 Polygenic risk scores | Privacy protected federated omic data analysis in multi-center studies with DataSHIELD" />
  
  
  



<meta name="date" content="2022-01-27" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="genome-wide-association-study-gwas-analysis.html"/>
<link rel="next" href="differential-gene-expression-dge-analysis.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script>
  function unrolltab(evt, divName, tabName) {
    var i, tabcontent, tablinks;
    tabcontent = document.getElementById(divName).getElementsByClassName("tabcontent");
    for (i = 0; i < tabcontent.length; i++) {
      tabcontent[i].style.display = "none";
    }
    tablinks = document.getElementById(divName).getElementsByClassName("tablinks");
    for (i = 0; i < tablinks.length; i++) {
      tablinks[i].className = tablinks[i].className.replace(" active", "");
    }
    document.getElementById(tabName).style.display = "block";
    evt.currentTarget.className += " active";
  }
</script>

<script>
  document.addEventListener("DOMContentLoaded", function(event) { 
    document.getElementById("gwas_results_1").click();
    document.getElementById("gwas_plot_1").click();
    document.getElementById("manhattan-plots-tab").click();
    document.getElementById("locuszoom-plots-tabs").click();
 });
</script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
/* Used with Pandoc 2.11+ new --citeproc when CSL is used */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="" target="blank">OmicSHIELD</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Welcome</a></li>
<li class="part"><span><b>I Preamble</b></span></li>
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#materials-to-read-beforehand"><i class="fa fa-check"></i><b>1.1</b> Materials to read beforehand</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#what-are-resources-a-very-simple-explanation-without-any-technicalities"><i class="fa fa-check"></i><b>1.2</b> What are “resources”: A very simple explanation without any technicalities</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#capabilities-of-omicshield"><i class="fa fa-check"></i><b>1.3</b> Capabilities of OmicSHIELD</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="Omic.html"><a href="Omic.html"><i class="fa fa-check"></i><b>2</b> Omic data analysis: types of implemented analyses</a></li>
<li class="part"><span><b>II Workflows</b></span></li>
<li class="chapter" data-level="3" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html"><i class="fa fa-check"></i><b>3</b> Genome wide association study (GWAS) analysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#single-cohort"><i class="fa fa-check"></i><b>3.1</b> Single cohort</a>
<ul>
<li class="chapter" data-level="3.1.1" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#connection-to-the-opal-server"><i class="fa fa-check"></i><b>3.1.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="3.1.2" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#assign-the-vcf-resources"><i class="fa fa-check"></i><b>3.1.2</b> Assign the VCF resources</a></li>
<li class="chapter" data-level="3.1.3" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#assign-the-phenotypes"><i class="fa fa-check"></i><b>3.1.3</b> Assign the phenotypes</a></li>
<li class="chapter" data-level="3.1.4" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#vcf_merge"><i class="fa fa-check"></i><b>3.1.4</b> Merge VCF (genotype) and phenotype information</a></li>
<li class="chapter" data-level="3.1.5" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#performing-the-gwas"><i class="fa fa-check"></i><b>3.1.5</b> Performing the GWAS</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#multi-cohorts"><i class="fa fa-check"></i><b>3.2</b> Multi cohorts</a>
<ul>
<li class="chapter" data-level="3.2.1" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#connection-to-the-opal-server-1"><i class="fa fa-check"></i><b>3.2.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="3.2.2" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#multi-vcf"><i class="fa fa-check"></i><b>3.2.2</b> Assign the VCF resources</a></li>
<li class="chapter" data-level="3.2.3" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#assign-the-phenotypes-1"><i class="fa fa-check"></i><b>3.2.3</b> Assign the phenotypes</a></li>
<li class="chapter" data-level="3.2.4" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#merge-vcf-genotype-and-phenotype-information"><i class="fa fa-check"></i><b>3.2.4</b> Merge VCF (genotype) and phenotype information</a></li>
<li class="chapter" data-level="3.2.5" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#performing-the-gwas-1"><i class="fa fa-check"></i><b>3.2.5</b> Performing a meta - GWAS</a></li>
<li class="chapter" data-level="3.2.6" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#gwas-plots"><i class="fa fa-check"></i><b>3.2.6</b> GWAS Plots</a></li>
<li class="chapter" data-level="3.2.7" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#meta-level-quality-control"><i class="fa fa-check"></i><b>3.2.7</b> Meta-level quality control</a></li>
<li class="chapter" data-level="3.2.8" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#meta-analysis"><i class="fa fa-check"></i><b>3.2.8</b> Meta-analysis</a></li>
<li class="chapter" data-level="3.2.9" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#ultra-fast-gwas"><i class="fa fa-check"></i><b>3.2.9</b> Ultra fast pooled GWAS analysis</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#post-omics"><i class="fa fa-check"></i><b>3.3</b> Post-omic analysis</a>
<ul>
<li class="chapter" data-level="3.3.1" data-path="genome-wide-association-study-gwas-analysis.html"><a href="genome-wide-association-study-gwas-analysis.html#enrichment-analysis"><i class="fa fa-check"></i><b>3.3.1</b> Enrichment analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html"><i class="fa fa-check"></i><b>4</b> Polygenic risk scores</a>
<ul>
<li class="chapter" data-level="4.1" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#definition"><i class="fa fa-check"></i><b>4.1</b> Definition</a></li>
<li class="chapter" data-level="4.2" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#study-configuration"><i class="fa fa-check"></i><b>4.2</b> Study configuration</a></li>
<li class="chapter" data-level="4.3" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#source-of-polygenic-scores"><i class="fa fa-check"></i><b>4.3</b> Source of polygenic scores</a></li>
<li class="chapter" data-level="4.4" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#connection-to-the-opal-server-2"><i class="fa fa-check"></i><b>4.4</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="4.5" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#assign-the-vcf-resources-1"><i class="fa fa-check"></i><b>4.5</b> Assign the VCF resources</a></li>
<li class="chapter" data-level="4.6" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#calculate-the-prs"><i class="fa fa-check"></i><b>4.6</b> Calculate the PRS</a>
<ul>
<li class="chapter" data-level="4.6.1" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#pgs-catalog-example"><i class="fa fa-check"></i><b>4.6.1</b> PGS Catalog Example</a></li>
<li class="chapter" data-level="4.6.2" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#custom-prs_table-example"><i class="fa fa-check"></i><b>4.6.2</b> Custom <code>prs_table</code> example</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="polygenic-risk-scores.html"><a href="polygenic-risk-scores.html#study-of-the-prs-results"><i class="fa fa-check"></i><b>4.7</b> Study of the PRS results</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html"><i class="fa fa-check"></i><b>5</b> Differential gene expression (DGE) analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#connection-to-the-opal-server-3"><i class="fa fa-check"></i><b>5.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="5.2" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#assign-the-rse-resource"><i class="fa fa-check"></i><b>5.2</b> Assign the RSE resource</a></li>
<li class="chapter" data-level="5.3" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#inspect-the-rse"><i class="fa fa-check"></i><b>5.3</b> Inspect the RSE</a></li>
<li class="chapter" data-level="5.4" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#pre-processing-for-rnaseq-data"><i class="fa fa-check"></i><b>5.4</b> Pre-processing for RNAseq data</a></li>
<li class="chapter" data-level="5.5" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#dge-analysis"><i class="fa fa-check"></i><b>5.5</b> DGE analysis</a></li>
<li class="chapter" data-level="5.6" data-path="differential-gene-expression-dge-analysis.html"><a href="differential-gene-expression-dge-analysis.html#surrogate-variable-analysis"><i class="fa fa-check"></i><b>5.6</b> Surrogate variable analysis</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html"><i class="fa fa-check"></i><b>6</b> Epigenome-wide association analysis (EWAS)</a>
<ul>
<li class="chapter" data-level="6.1" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#initial-steps-for-all-use-cases"><i class="fa fa-check"></i><b>6.1</b> Initial steps for all use cases</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#connection-to-the-opal-server-4"><i class="fa fa-check"></i><b>6.1.1</b> Connection to the Opal server</a></li>
<li class="chapter" data-level="6.1.2" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#assign-the-expressionset-resource"><i class="fa fa-check"></i><b>6.1.2</b> Assign the <code>ExpressionSet</code> resource</a></li>
<li class="chapter" data-level="6.1.3" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#ewas-inspect"><i class="fa fa-check"></i><b>6.1.3</b> Inspect the <code>ExpressionSet</code></a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#single-cpg-pooled-analysis"><i class="fa fa-check"></i><b>6.2</b> Single CpG pooled analysis</a></li>
<li class="chapter" data-level="6.3" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#multiple-cpgs-pooled-analysis"><i class="fa fa-check"></i><b>6.3</b> Multiple CpGs pooled analysis</a></li>
<li class="chapter" data-level="6.4" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#full-genome-meta-analysis"><i class="fa fa-check"></i><b>6.4</b> Full genome meta-analysis</a></li>
<li class="chapter" data-level="6.5" data-path="epigenome-wide-association-analysis-ewas.html"><a href="epigenome-wide-association-analysis-ewas.html#adjusting-for-surrogate-variables"><i class="fa fa-check"></i><b>6.5</b> Adjusting for surrogate variables</a></li>
</ul></li>
<li class="part"><span><b>III Technical details</b></span></li>
<li class="chapter" data-level="7" data-path="gwas.html"><a href="gwas.html"><i class="fa fa-check"></i><b>7</b> GWAS</a></li>
<li class="chapter" data-level="8" data-path="internal-structure-of-the-prs-functionality.html"><a href="internal-structure-of-the-prs-functionality.html"><i class="fa fa-check"></i><b>8</b> Internal structure of the PRS functionality</a>
<ul>
<li class="chapter" data-level="8.1" data-path="internal-structure-of-the-prs-functionality.html"><a href="internal-structure-of-the-prs-functionality.html#client"><i class="fa fa-check"></i><b>8.1</b> Client</a></li>
<li class="chapter" data-level="8.2" data-path="internal-structure-of-the-prs-functionality.html"><a href="internal-structure-of-the-prs-functionality.html#server-resolver"><i class="fa fa-check"></i><b>8.2</b> Server resolver</a></li>
<li class="chapter" data-level="8.3" data-path="internal-structure-of-the-prs-functionality.html"><a href="internal-structure-of-the-prs-functionality.html#server-analysis"><i class="fa fa-check"></i><b>8.3</b> Server analysis</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="compression-of-gds-files-vs.-performance.html"><a href="compression-of-gds-files-vs.-performance.html"><i class="fa fa-check"></i><b>9</b> Compression of GDS files vs. performance</a></li>
<li class="part"><span><b>IV Appendix</b></span></li>
<li class="chapter" data-level="10" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>10</b> Session info</a></li>
<li class="chapter" data-level="11" data-path="contributos.html"><a href="contributos.html"><i class="fa fa-check"></i><b>11</b> Contributos</a></li>
<li class="part"><span><b>V References</b></span></li>
<li class="chapter" data-level="12" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i><b>12</b> References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Privacy protected federated omic data analysis in multi-center studies with DataSHIELD</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="polygenic-risk-scores" class="section level1" number="4">
<h1><span class="header-section-number">4</span> Polygenic risk scores</h1>
<table>
<colgroup>
<col style="width: 100%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">
<p>
⚠️ RESOURCES USED ALONG THIS SECTION
</p>
</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">
<p>
From <a href="https://opal-demo.obiba.org/">https://opal-demo.obiba.org/</a> :
</p>
</td>
</tr>
<tr class="even">
<td style="text-align: left;">
<table>
<thead>
<tr class="header">
<th>
<p>
STUDY
</p>
</th>
<th>
<p>
TABLE
</p>
</th>
<th>
<p>
PROFILE
</p>
</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>
<p>
cohort1
</p>
</td>
<td>
<p>
GWAS.chr{1..21}A
</p>
</td>
<td>
<p>
omics
</p>
</td>
</tr>
<tr class="odd">
<td>
<p>
cohort2
</p>
</td>
<td>
<p>
GWAS.chr{1..21}B
</p>
</td>
<td>
<p>
omics
</p>
</td>
</tr>
<tr class="odd">
<td>
<p>
cohort3
</p>
</td>
<td>
<p>
GWAS.chr{1..21}C
</p>
</td>
<td>
<p>
omics
</p>
</td>
</tr>
<tr class="odd">
<td>
<p>
cohort1
</p>
</td>
<td>
<p>
GWAS.ega_phenotypes_1
</p>
</td>
<td>
<p>
omics
</p>
</td>
</tr>
<tr class="odd">
<td>
<p>
cohort2
</p>
</td>
<td>
<p>
GWAS.ega_phenotypes_2
</p>
</td>
<td>
<p>
omics
</p>
</td>
</tr>
<tr class="odd">
<td>
<p>
cohort3
</p>
</td>
<td>
<p>
GWAS.ega_phenotypes_3
</p>
</td>
<td>
<p>
omics
</p>
</td>
</tr>
<tr 
</tbody></table></td></tr>
</tbody>
</table>
<div id="definition" class="section level2" number="4.1">
<h2><span class="header-section-number">4.1</span> Definition</h2>
<p>By checking for specific variants and quantifying them, a researcher can extract the polygenic risk score (PRS) of an individual, which translates to an associated risk of the individual versus a concrete phenotype. A definition of the term reads</p>
<blockquote>
<p>“A weighted sum of the number of risk alleles carried by an individual, where the risk alleles and their weights are defined by the loci and their measured effects as detected by genome wide association studies.” (Extracted from <span class="citation"><a href="#ref-torkamani2018personal" role="doc-biblioref">Torkamani, Wineinger, and Topol</a> (<a href="#ref-torkamani2018personal" role="doc-biblioref">2018</a>)</span>)</p>
</blockquote>
<p>The use of PRS markers is very popular nowadays and it has been proved to be a good tool to asses associated risks <span class="citation"><a href="#ref-escott2017polygenic" role="doc-biblioref">Escott-Price et al.</a> (<a href="#ref-escott2017polygenic" role="doc-biblioref">2017</a>)</span>, <span class="citation"><a href="#ref-forgetta2020development" role="doc-biblioref">Forgetta et al.</a> (<a href="#ref-forgetta2020development" role="doc-biblioref">2020</a>)</span>, <span class="citation"><a href="#ref-kramer2020breast" role="doc-biblioref">Kramer et al.</a> (<a href="#ref-kramer2020breast" role="doc-biblioref">2020</a>)</span>.</p>
<p>When calculating the PRS using DataSHIELD, the actual scores are not returned, as it would not be GDPR compliant to be able to determine the risk of an individual versus a certain disease. The actual results are stored on the servers and optionally can be merged to the phenotypes table. This allows further study of the PRS using summaries or fitting pooled GLM models to assess relationships between phenotypes and PRS scores.</p>
</div>
<div id="study-configuration" class="section level2" number="4.2">
<h2><span class="header-section-number">4.2</span> Study configuration</h2>
<p>The PRS use case will be set up as a multi-cohort, using the same dataset as the multi-cohort GWAS (same VCF resources). A single cohort PRS could be performed analogously using the single cohort example of the GWAS.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:prsCohortImage"></span>
<img src="fig/prs.PNG" alt="Proposed infrastructure to perform PRS studies." width="918" />
<p class="caption">
Figure 4.1: Proposed infrastructure to perform PRS studies.
</p>
</div>
</div>
<div id="source-of-polygenic-scores" class="section level2" number="4.3">
<h2><span class="header-section-number">4.3</span> Source of polygenic scores</h2>
<p>There is a public repository of polygenic scores available at <a href="https://www.pgscatalog.org/" class="uri">https://www.pgscatalog.org/</a>, on this website there are 2155 different polygenic scores (January 2022) that have been extracted from curated literature. All those scores refer to 526 different traits. Due the relevance of this project, we made sure to include a way of sourcing it’s contents automatically.</p>
<p>There are two ways of stating the SNPs of interest and their weights in order to calculate the PRS.</p>
<ul>
<li><p>Provide a <a href="https://www.pgscatalog.org/browse/scores/">PGS Catalog</a> ‘Polygenic Score ID &amp; Name.’ If this option is in use, the SNPs and beta weights will be downloaded from the PGS Catalog and will be used to calculate the PRS. (Example: <code>PGS000008</code>)</p></li>
<li><p>Providing a <code>prs_table</code> (SNPs of interest) table. The <code>prs_table</code> table has to have a defined structure and column names in order to be understood by this function. This table can be formulated using two schemas:</p>
<ul>
<li>Scheme 1: Provide SNP positions. Use the column names: <code>chr_name</code>, <code>chr_position</code>, <code>effect_allele</code>, <code>reference_allele</code>, <code>effect_weight</code>.</li>
</ul>
<table>
<thead>
<tr class="header">
<th>chr_name</th>
<th>chr_position</th>
<th>effect_allele</th>
<th>reference_allele</th>
<th>effect_weight</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>100880328</td>
<td>T</td>
<td>A</td>
<td>0.0373</td>
</tr>
<tr class="even">
<td>1</td>
<td>121287994</td>
<td>G</td>
<td>A</td>
<td>-0.0673</td>
</tr>
<tr class="odd">
<td>6</td>
<td>152023191</td>
<td>A</td>
<td>G</td>
<td>0.0626</td>
</tr>
<tr class="even">
<td>…</td>
<td>…</td>
<td>…</td>
<td>…</td>
<td>…</td>
</tr>
</tbody>
</table>
<ul>
<li>Scheme 2: Provide SNP id’s. Use the column names: <code>rsID</code>, <code>effect_allele</code>, <code>reference_allele</code>, <code>effect_weight</code>.</li>
</ul>
<table>
<thead>
<tr class="header">
<th>rdID</th>
<th>effect_allele</th>
<th>reference_allele</th>
<th>effect_weight</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>rs11206510</td>
<td>T</td>
<td>C</td>
<td>0.0769</td>
</tr>
<tr class="even">
<td>rs4773144</td>
<td>G</td>
<td>A</td>
<td>0.0676</td>
</tr>
<tr class="odd">
<td>rs17514846</td>
<td>A</td>
<td>C</td>
<td>0.0487</td>
</tr>
<tr class="even">
<td>…</td>
<td>…</td>
<td>…</td>
<td>…</td>
</tr>
</tbody>
</table></li>
</ul>
<p>The <code>effect_weight</code> have to be the betas (log(OR)).</p>
<p>Please note when using a custom <code>prs_table</code> table that it is much faster to use the Schema 1, as the subset of the VCF files is miles faster to do using chromosome name and positions rather than SNP id’s.</p>
</div>
<div id="connection-to-the-opal-server-2" class="section level2" number="4.4">
<h2><span class="header-section-number">4.4</span> Connection to the Opal server</h2>
<p>We have to create an Opal connection object to the cohort server. We do that using the following functions.</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="polygenic-risk-scores.html#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(<span class="st">&#39;DSI&#39;</span>)</span>
<span id="cb80-2"><a href="polygenic-risk-scores.html#cb80-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(<span class="st">&#39;DSOpal&#39;</span>)</span>
<span id="cb80-3"><a href="polygenic-risk-scores.html#cb80-3" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(<span class="st">&#39;dsBaseClient&#39;</span>)</span>
<span id="cb80-4"><a href="polygenic-risk-scores.html#cb80-4" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(<span class="st">&#39;dsOmicsClient&#39;</span>)</span>
<span id="cb80-5"><a href="polygenic-risk-scores.html#cb80-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-6"><a href="polygenic-risk-scores.html#cb80-6" aria-hidden="true" tabindex="-1"></a>builder <span class="ot">&lt;-</span> DSI<span class="sc">::</span><span class="fu">newDSLoginBuilder</span>()</span>
<span id="cb80-7"><a href="polygenic-risk-scores.html#cb80-7" aria-hidden="true" tabindex="-1"></a>builder<span class="sc">$</span><span class="fu">append</span>(<span class="at">server =</span> <span class="st">&quot;cohort1&quot;</span>, <span class="at">url =</span> <span class="st">&quot;https://opal-demo.obiba.org/&quot;</span>,</span>
<span id="cb80-8"><a href="polygenic-risk-scores.html#cb80-8" aria-hidden="true" tabindex="-1"></a>               <span class="at">user =</span>  <span class="st">&quot;dsuser&quot;</span>, <span class="at">password =</span> <span class="st">&quot;P@ssw0rd&quot;</span>,</span>
<span id="cb80-9"><a href="polygenic-risk-scores.html#cb80-9" aria-hidden="true" tabindex="-1"></a>               <span class="at">driver =</span> <span class="st">&quot;OpalDriver&quot;</span>, <span class="at">profile =</span> <span class="st">&quot;omics&quot;</span>)</span>
<span id="cb80-10"><a href="polygenic-risk-scores.html#cb80-10" aria-hidden="true" tabindex="-1"></a>builder<span class="sc">$</span><span class="fu">append</span>(<span class="at">server =</span> <span class="st">&quot;cohort2&quot;</span>, <span class="at">url =</span> <span class="st">&quot;https://opal-demo.obiba.org/&quot;</span>,</span>
<span id="cb80-11"><a href="polygenic-risk-scores.html#cb80-11" aria-hidden="true" tabindex="-1"></a>               <span class="at">user =</span>  <span class="st">&quot;dsuser&quot;</span>, <span class="at">password =</span> <span class="st">&quot;P@ssw0rd&quot;</span>,</span>
<span id="cb80-12"><a href="polygenic-risk-scores.html#cb80-12" aria-hidden="true" tabindex="-1"></a>               <span class="at">driver =</span> <span class="st">&quot;OpalDriver&quot;</span>, <span class="at">profile =</span> <span class="st">&quot;omics&quot;</span>)</span>
<span id="cb80-13"><a href="polygenic-risk-scores.html#cb80-13" aria-hidden="true" tabindex="-1"></a>builder<span class="sc">$</span><span class="fu">append</span>(<span class="at">server =</span> <span class="st">&quot;cohort3&quot;</span>, <span class="at">url =</span> <span class="st">&quot;https://opal-demo.obiba.org/&quot;</span>,</span>
<span id="cb80-14"><a href="polygenic-risk-scores.html#cb80-14" aria-hidden="true" tabindex="-1"></a>               <span class="at">user =</span>  <span class="st">&quot;dsuser&quot;</span>, <span class="at">password =</span> <span class="st">&quot;P@ssw0rd&quot;</span>,</span>
<span id="cb80-15"><a href="polygenic-risk-scores.html#cb80-15" aria-hidden="true" tabindex="-1"></a>               <span class="at">driver =</span> <span class="st">&quot;OpalDriver&quot;</span>, <span class="at">profile =</span> <span class="st">&quot;omics&quot;</span>)</span>
<span id="cb80-16"><a href="polygenic-risk-scores.html#cb80-16" aria-hidden="true" tabindex="-1"></a>logindata <span class="ot">&lt;-</span> builder<span class="sc">$</span><span class="fu">build</span>()</span>
<span id="cb80-17"><a href="polygenic-risk-scores.html#cb80-17" aria-hidden="true" tabindex="-1"></a>conns <span class="ot">&lt;-</span> DSI<span class="sc">::</span><span class="fu">datashield.login</span>(<span class="at">logins =</span> logindata)</span></code></pre></div>
</div>
<div id="assign-the-vcf-resources-1" class="section level2" number="4.5">
<h2><span class="header-section-number">4.5</span> Assign the VCF resources</h2>
<p>We assign the VCF and phenotypes resources.</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="polygenic-risk-scores.html#cb81-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Cohort 1 resources</span></span>
<span id="cb81-2"><a href="polygenic-risk-scores.html#cb81-2" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">21</span>, <span class="cf">function</span>(x){</span>
<span id="cb81-3"><a href="polygenic-risk-scores.html#cb81-3" aria-hidden="true" tabindex="-1"></a>  DSI<span class="sc">::</span><span class="fu">datashield.assign.resource</span>(conns[<span class="dv">1</span>], <span class="fu">paste0</span>(<span class="st">&quot;chr&quot;</span>, x), <span class="fu">paste0</span>(<span class="st">&quot;GWAS.chr&quot;</span>, x,<span class="st">&quot;A&quot;</span>))</span>
<span id="cb81-4"><a href="polygenic-risk-scores.html#cb81-4" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb81-5"><a href="polygenic-risk-scores.html#cb81-5" aria-hidden="true" tabindex="-1"></a>DSI<span class="sc">::</span><span class="fu">datashield.assign.resource</span>(conns[<span class="dv">1</span>], <span class="st">&quot;phenotypes_resource&quot;</span>, <span class="st">&quot;GWAS.ega_phenotypes_1&quot;</span>)</span>
<span id="cb81-6"><a href="polygenic-risk-scores.html#cb81-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb81-7"><a href="polygenic-risk-scores.html#cb81-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Cohort 2 resources</span></span>
<span id="cb81-8"><a href="polygenic-risk-scores.html#cb81-8" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">21</span>, <span class="cf">function</span>(x){</span>
<span id="cb81-9"><a href="polygenic-risk-scores.html#cb81-9" aria-hidden="true" tabindex="-1"></a>  DSI<span class="sc">::</span><span class="fu">datashield.assign.resource</span>(conns[<span class="dv">2</span>], <span class="fu">paste0</span>(<span class="st">&quot;chr&quot;</span>, x), <span class="fu">paste0</span>(<span class="st">&quot;GWAS.chr&quot;</span>, x,<span class="st">&quot;B&quot;</span>))</span>
<span id="cb81-10"><a href="polygenic-risk-scores.html#cb81-10" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb81-11"><a href="polygenic-risk-scores.html#cb81-11" aria-hidden="true" tabindex="-1"></a>DSI<span class="sc">::</span><span class="fu">datashield.assign.resource</span>(conns[<span class="dv">2</span>], <span class="st">&quot;phenotypes_resource&quot;</span>, <span class="st">&quot;GWAS.ega_phenotypes_2&quot;</span>)</span>
<span id="cb81-12"><a href="polygenic-risk-scores.html#cb81-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb81-13"><a href="polygenic-risk-scores.html#cb81-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Cohort 3 resources</span></span>
<span id="cb81-14"><a href="polygenic-risk-scores.html#cb81-14" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">21</span>, <span class="cf">function</span>(x){</span>
<span id="cb81-15"><a href="polygenic-risk-scores.html#cb81-15" aria-hidden="true" tabindex="-1"></a>  DSI<span class="sc">::</span><span class="fu">datashield.assign.resource</span>(conns[<span class="dv">3</span>], <span class="fu">paste0</span>(<span class="st">&quot;chr&quot;</span>, x), <span class="fu">paste0</span>(<span class="st">&quot;GWAS.chr&quot;</span>, x,<span class="st">&quot;C&quot;</span>))</span>
<span id="cb81-16"><a href="polygenic-risk-scores.html#cb81-16" aria-hidden="true" tabindex="-1"></a>  })</span>
<span id="cb81-17"><a href="polygenic-risk-scores.html#cb81-17" aria-hidden="true" tabindex="-1"></a>DSI<span class="sc">::</span><span class="fu">datashield.assign.resource</span>(conns[<span class="dv">3</span>], <span class="st">&quot;phenotypes_resource&quot;</span>, <span class="st">&quot;GWAS.ega_phenotypes_3&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="polygenic-risk-scores.html#cb82-1" aria-hidden="true" tabindex="-1"></a>DSI<span class="sc">::</span><span class="fu">datashield.assign.expr</span>(<span class="at">conns =</span> conns, <span class="at">symbol =</span> <span class="st">&quot;phenotypes&quot;</span>,</span>
<span id="cb82-2"><a href="polygenic-risk-scores.html#cb82-2" aria-hidden="true" tabindex="-1"></a>                            <span class="at">expr =</span> <span class="fu">as.symbol</span>(<span class="st">&quot;as.resource.data.frame(phenotypes_resource)&quot;</span>))</span></code></pre></div>
<p>There is one crucial difference between this analysis and the GWAS, here we don’t need to resolve the resources. This is for memory efficiency purposes. When we calculate a PRS, we are only interested in a small selection of SNPs, for that reason the (potentially) huge VCF files will be trimmed before being resolved, so only the relevant SNPs are loaded in the memory. For that reason the resolving of the resources is delegated to the <code>ds.PRS</code> function, which will perform this trimming according to the SNPs of interest yielded by the PGS catalog query or the <code>prs_table</code> table.</p>
</div>
<div id="calculate-the-prs" class="section level2" number="4.6">
<h2><span class="header-section-number">4.6</span> Calculate the PRS</h2>
<p>The PRS results will be added to the phenotypes table to be later studied using a pooled GLM. To do that, the following arguments of the function <code>ds.PRS</code> will be used:</p>
<ul>
<li><code>table</code>: Name of the table where the results will be merged.</li>
<li><code>table_id_column</code>: Number of the column in the <code>table</code> table that contains the individuals ID to merge the results from the PRS.</li>
<li><code>table_prs_name</code>: Name to append to the PRS results. It will take the structure <code>prs_table_prs_name</code> and <code>prs_nw_table_prs_name</code>.</li>
</ul>
<div id="pgs-catalog-example" class="section level3" number="4.6.1">
<h3><span class="header-section-number">4.6.1</span> PGS Catalog Example</h3>
<p>For this example we will calculate the PRS associated to the trait ‘HDL Cholesterol’ and ‘Thyroid cancer,’ which corresponds to the PGS ID <a href="https://www.pgscatalog.org/score/PGS000660/">PGS000660</a> and <a href="https://www.pgscatalog.org/score/PGS000636/">PGS000636</a></p>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="polygenic-risk-scores.html#cb83-1" aria-hidden="true" tabindex="-1"></a>resources <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;chr&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">21</span>)</span>
<span id="cb83-2"><a href="polygenic-risk-scores.html#cb83-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(resources) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;chr&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">21</span>)</span>
<span id="cb83-3"><a href="polygenic-risk-scores.html#cb83-3" aria-hidden="true" tabindex="-1"></a>resources <span class="ot">&lt;-</span> <span class="fu">as.list</span>(resources)</span>
<span id="cb83-4"><a href="polygenic-risk-scores.html#cb83-4" aria-hidden="true" tabindex="-1"></a><span class="co"># HDL cholesterol</span></span>
<span id="cb83-5"><a href="polygenic-risk-scores.html#cb83-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ds.PRS</span>(<span class="at">resources =</span> resources, <span class="at">pgs_id =</span> <span class="st">&quot;PGS000660&quot;</span>, </span>
<span id="cb83-6"><a href="polygenic-risk-scores.html#cb83-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">table =</span> <span class="st">&quot;phenotypes&quot;</span>, <span class="at">table_id_column =</span> <span class="st">&quot;subject_id&quot;</span>)</span>
<span id="cb83-7"><a href="polygenic-risk-scores.html#cb83-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Thyroid cancer</span></span>
<span id="cb83-8"><a href="polygenic-risk-scores.html#cb83-8" aria-hidden="true" tabindex="-1"></a><span class="fu">ds.PRS</span>(<span class="at">resources =</span> resources, <span class="at">pgs_id =</span> <span class="st">&quot;PGS000636&quot;</span>, </span>
<span id="cb83-9"><a href="polygenic-risk-scores.html#cb83-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">table =</span> <span class="st">&quot;phenotypes&quot;</span>, <span class="at">table_id_column =</span> <span class="st">&quot;subject_id&quot;</span>)</span></code></pre></div>
<p>We can check that the results have been correctly added to the <code>phenotypes</code> table.</p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="polygenic-risk-scores.html#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tail</span>(<span class="fu">ds.colnames</span>(<span class="st">&#39;phenotypes&#39;</span>)[[<span class="dv">1</span>]])</span></code></pre></div>
<pre><code>[1] &quot;prs_PGS000660&quot;    &quot;prs_nw_PGS000660&quot; &quot;n_snps_PGS000660&quot; &quot;prs_PGS000636&quot;   
[5] &quot;prs_nw_PGS000636&quot; &quot;n_snps_PGS000636&quot;</code></pre>
</div>
<div id="custom-prs_table-example" class="section level3" number="4.6.2">
<h3><span class="header-section-number">4.6.2</span> Custom <code>prs_table</code> example</h3>
<p>Here we will illustrate a use case with a custom <code>prs_table</code>, please note that the proposed SNPs of interest and betas are totally arbitrary.</p>
<p>We will first assemble the <code>prs_table</code> table following the indications found on the Details section of <code>?ds.PRS</code> (Schema 1).</p>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="polygenic-risk-scores.html#cb86-1" aria-hidden="true" tabindex="-1"></a>prs_table <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">chr_name =</span> <span class="fu">c</span>(<span class="dv">9</span>, <span class="dv">9</span>, <span class="dv">9</span>, <span class="dv">9</span>, <span class="dv">1</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">2</span>),</span>
<span id="cb86-2"><a href="polygenic-risk-scores.html#cb86-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">chr_position =</span> <span class="fu">c</span>(<span class="dv">177034</span>, <span class="dv">161674</span>, <span class="dv">140859</span>, <span class="dv">119681</span>, </span>
<span id="cb86-3"><a href="polygenic-risk-scores.html#cb86-3" aria-hidden="true" tabindex="-1"></a>                                   <span class="dv">60351</span>, <span class="dv">825410</span>, <span class="dv">154629</span>, <span class="dv">145710</span>),</span>
<span id="cb86-4"><a href="polygenic-risk-scores.html#cb86-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">effect_allele =</span> <span class="fu">c</span>(<span class="st">&quot;T&quot;</span>, <span class="st">&quot;C&quot;</span>, <span class="st">&quot;G&quot;</span>, <span class="st">&quot;C&quot;</span>, <span class="st">&quot;G&quot;</span>, <span class="st">&quot;G&quot;</span>, <span class="st">&quot;T&quot;</span>, <span class="st">&quot;A&quot;</span>),</span>
<span id="cb86-5"><a href="polygenic-risk-scores.html#cb86-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">reference_allele =</span> <span class="fu">c</span>(<span class="st">&quot;G&quot;</span>, <span class="st">&quot;T&quot;</span>, <span class="st">&quot;A&quot;</span>, <span class="st">&quot;T&quot;</span>, <span class="st">&quot;A&quot;</span>, <span class="st">&quot;A&quot;</span>, <span class="st">&quot;G&quot;</span>, <span class="st">&quot;G&quot;</span>),</span>
<span id="cb86-6"><a href="polygenic-risk-scores.html#cb86-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">effect_weight =</span> <span class="fu">c</span>(<span class="fl">0.5</span>, <span class="fl">0.2</span>, <span class="fl">0.1</span>, <span class="fl">0.2</span>, <span class="fl">0.1</span>, <span class="fl">0.3</span>, <span class="fl">0.1</span>, <span class="fl">0.8</span>))</span></code></pre></div>
<p>Now we can pass the custom <code>prs_table</code> to the PRS function to retrieve the aggregated results.</p>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="polygenic-risk-scores.html#cb87-1" aria-hidden="true" tabindex="-1"></a>resources <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;chr&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">21</span>)</span>
<span id="cb87-2"><a href="polygenic-risk-scores.html#cb87-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(resources) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;chr&quot;</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">21</span>)</span>
<span id="cb87-3"><a href="polygenic-risk-scores.html#cb87-3" aria-hidden="true" tabindex="-1"></a>resources <span class="ot">&lt;-</span> <span class="fu">as.list</span>(resources)</span>
<span id="cb87-4"><a href="polygenic-risk-scores.html#cb87-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ds.PRS</span>(<span class="at">resources =</span> resources, <span class="at">prs_table =</span> prs_table, </span>
<span id="cb87-5"><a href="polygenic-risk-scores.html#cb87-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">table =</span> <span class="st">&quot;phenotypes&quot;</span>, <span class="at">table_id_column =</span> <span class="st">&quot;subject_id&quot;</span>, </span>
<span id="cb87-6"><a href="polygenic-risk-scores.html#cb87-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">table_prs_name =</span> <span class="st">&quot;custom_prs_table_results&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="study-of-the-prs-results" class="section level2" number="4.7">
<h2><span class="header-section-number">4.7</span> Study of the PRS results</h2>
<p>We can fit pooled GLM models using the function <code>ds.glm</code>. In this use case we will fit a model to relate the <code>HDL.cholesterol</code> to <code>prs_PGS000660</code> adjusted by <code>sex</code>. The same will be done with the <code>prs_PGS000636</code>.</p>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="polygenic-risk-scores.html#cb88-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ds.glm</span>(<span class="at">formula =</span> <span class="st">&quot;hdl_cholesterol ~ prs_PGS000660 + sex&quot;</span>, </span>
<span id="cb88-2"><a href="polygenic-risk-scores.html#cb88-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">family =</span> <span class="st">&quot;gaussian&quot;</span>, <span class="at">data =</span> <span class="st">&quot;phenotypes&quot;</span>)<span class="sc">$</span>coefficients</span></code></pre></div>
<pre><code>                  Estimate  Std. Error    z-value      p-value   low0.95CI
(Intercept)    1.576448314 0.178438076  8.8347081 1.003600e-18  1.22671611
prs_PGS000660 -0.003678982 0.009899804 -0.3716217 7.101746e-01 -0.02308224
sexmale        0.002311580 0.016140511  0.1432161 8.861195e-01 -0.02932324
              high0.95CI
(Intercept)   1.92618052
prs_PGS000660 0.01572428
sexmale       0.03394640</code></pre>
<div class="sourceCode" id="cb90"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb90-1"><a href="polygenic-risk-scores.html#cb90-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ds.glm</span>(<span class="at">formula =</span> <span class="st">&quot;hdl_cholesterol ~ prs_PGS000636 + sex&quot;</span>, </span>
<span id="cb90-2"><a href="polygenic-risk-scores.html#cb90-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">family =</span> <span class="st">&quot;gaussian&quot;</span>, <span class="at">data =</span> <span class="st">&quot;phenotypes&quot;</span>)<span class="sc">$</span>coefficients</span></code></pre></div>
<pre><code>                  Estimate Std. Error    z-value   p-value   low0.95CI
(Intercept)    1.522002437 0.01614425 94.2751901 0.0000000  1.49036028
prs_PGS000636 -0.066872228 0.06553929 -1.0203380 0.3075682 -0.19532688
sexmale        0.002416221 0.01613787  0.1497237 0.8809826 -0.02921342
              high0.95CI
(Intercept)   1.55364459
prs_PGS000636 0.06158242
sexmale       0.03404586</code></pre>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="polygenic-risk-scores.html#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="fu">datashield.logout</span>(conns)</span></code></pre></div>

</div>
</div>
<h3> References</h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-escott2017polygenic" class="csl-entry">
Escott-Price, Valentina, Amanda J Myers, Matt Huentelman, and John Hardy. 2017. <span>“Polygenic Risk Score Analysis of Pathologically Confirmed Alzheimer Disease.”</span> <em>Annals of Neurology</em> 82 (2): 311–14.
</div>
<div id="ref-forgetta2020development" class="csl-entry">
Forgetta, Vincenzo, Julyan Keller-Baruch, Marie Forest, Audrey Durand, Sahir Bhatnagar, John P Kemp, Maria Nethander, et al. 2020. <span>“Development of a Polygenic Risk Score to Improve Screening for Fracture Risk: A Genetic Risk Prediction Study.”</span> <em>PLoS Medicine</em> 17 (7): e1003152.
</div>
<div id="ref-kramer2020breast" class="csl-entry">
Kramer, Iris, Maartje J Hooning, Nasim Mavaddat, Michael Hauptmann, Renske Keeman, Ewout W Steyerberg, Daniele Giardiello, et al. 2020. <span>“Breast Cancer Polygenic Risk Score and Contralateral Breast Cancer Risk.”</span> <em>The American Journal of Human Genetics</em> 107 (5): 837–48.
</div>
<div id="ref-torkamani2018personal" class="csl-entry">
Torkamani, Ali, Nathan E Wineinger, and Eric J Topol. 2018. <span>“The Personal and Clinical Utility of Polygenic Risk Scores.”</span> <em>Nature Reviews Genetics</em> 19 (9): 581–90.
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="genome-wide-association-study-gwas-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="differential-gene-expression-dge-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": null,
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["OmicSHIELD_bookdown.pdf", "OmicSHIELD_bookdown.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
